zebra-network/src/peer_set/initialize.rs

```diff
@@ -109,8 +111,54 @@ where
 {
     let (tcp_listener, listen_addr) = open_listener(&config.clone()).await;
 
-    let (address_book, address_book_updater, address_metrics, address_book_updater_guard) =
-        AddressBookUpdater::spawn(&config, listen_addr);
+    let (
+        address_book,
+        bans_receiver,
+        address_book_updater,
+        address_metrics,
+        address_book_updater_guard,
+    ) = AddressBookUpdater::spawn(&config, listen_addr);
+
+    let (misbehavior_tx, mut misbehavior_rx) = mpsc::channel(
+        3 * config
+            .peerset_total_connection_limit()
+            .max(MIN_CHANNEL_SIZE),
+    );
+
+    let misbehaviour_updater = address_book_updater.clone();
+    tokio::spawn(
+        async move {
+            let mut misbehaviors: HashMap<PeerSocketAddr, u32> = HashMap::new();
+            let mut flush_timer =
+                IntervalStream::new(tokio::time::interval(Duration::from_secs(60)));
```

Comment:
It will be good to explain why a 1 minute interval is used here.

Reply:
This one's not really important until/unless we add finer-grained misbehaviour scores. Added a comment explaining why it's there in https://github.com/ZcashFoundation/zebra/pull/9201/commits/88e0c9c0405ee533f6308394928a3f8de9f161d0.